internal/storage/fake/bucket.go

```diff
@@ -992,29 +1042,33 @@ func (b *bucket) RenameFolder(ctx context.Context, folderName string, destinatio
 		return nil, err
 	}
 
-	dst := b.folders[srcIndex]
-	dst.Name = destinationFolderId
+	// Find all folders starting with the given prefix and update their names.
+	for i := range b.folders {
+		if strings.HasPrefix(b.folders[i].Name, folderName) {
```

Comment:
Rename folder is for a specific folder right. why are you doing this prefix check?

Reply:
Let's say we have the following structure:
```
A/B
A/a.txt
A/C/d.txt
```
If we rename "A" to "A_new", we need to check for the prefix "A/" in both the folder and object arrays to rename "A/B" to "A_new/B" and "A/a.txt" to "A_new/a.txt".

Reply:
Got it. It is to update the child directories.